modules/ingester/flush.go

```diff
@@ -283,6 +305,8 @@ func (i *Ingester) handleFlush(userID string, blockID uuid.UUID) (retry bool, er
 		metricFlushDuration.Observe(time.Since(start).Seconds())
 		metricFlushSize.Observe(float64(block.BlockMeta().Size))
 		if err != nil {
+			ext.Error.Set(sp, true)
+			sp.LogFields(otlog.Error(err))
```

Comment:
should we put this under a `defer()` block and check for err?

Reply:
That requires extreme care with scoping of `err`; it's not something I would naturally do.